### ISSUE TITLE
ES spec violation

### ISSUE THREAD

**Issue by user-982634** [babel/babel#7993]
## Bug Report

**Current Behavior**

The parser throws a SyntaxError

**Input Code**

This is a valid JavaScript code according to the latest ES standard:

```js
a=>{}
/\//
```

**Expected behavior/code**

No SyntaxError should be thrown

**Babel Configuration (.babelrc, package.json, cli command)**

Not sure, I'm using the [online babel parser](https://babeljs.io/)

**Possible Solution**

`’\_(ツ)_/‘`

**Additional context/Screenshots**

![](https://user-images.githubusercontent.com/39399001/40291616-4b008a2c-5cc6-11e8-848a-dd40bac67ed6.png)

**Comment by xtuc** [comment]
Closing, out of scope for now.

**Comment by nicolo-ribaudo** [comment]
@danez Is this fixed by your context PR?

**Comment by danez** [comment]
I do not think this is valid. I tried this in acorn, flow, esprima, chrome and node. All parsers fail the same way.

```
/\//
 ^

SyntaxError: Invalid or unexpected token
```

**Comment by user-2699515475722898** [comment]
@danez I'm sorry to disapoint you, but this is valid syntax according to the latest EcmaScript specification. I sincerely have no idea how you are testing the script. It works in Chrome, Node.js and Firefox. Probably you concatenated the two lines into single line. The key is that the new line is necessary. Also see [this](https://tio.run/##y0osSyxOLsosKNHNy09J/f8/0dauupZLP0Zfn4srOT@vOD8nVS8nP11DPSwxJzPFXl3z/38A) online Node.js interpreter that properly parses this.

Please reopen this and remove the invalid label. Thanks.

**Comment by danez** [comment]
Ohh, I'm not sure what happened but i was testing this:
```
a=>[]
/\//
```

with curly brackets yes it will be fixed with #8972
